### PR TITLE
[Bug-fix] fix vec::reverse

### PR DESCRIPTION
**Pull request by zbtzbtzbt**
## Proposed changes
```
mysql> select reverse('美团和和阿斯顿ab');
+------------------------------------+
| reverse('美团和和阿斯顿ab')        |
+------------------------------------+
| �ab�顿斘�和和团美                     |
+------------------------------------+
1 row in set (0.01 sec)


mysql> select reverse('美团和和ab');
+---------------------------+
| reverse('美团和和ab')     |
+---------------------------+
| �ab��和团美                  |
+---------------------------+
1 row in set (0.01 sec)

```
After
```
mysql> select reverse('美团和和阿斯顿ab');
+------------------------------------+
| reverse('美团和和阿斯顿ab')        |
+------------------------------------+
| ba顿斯阿和和团美                   |
+------------------------------------+
1 row in set (0.01 sec)

mysql> select reverse('美团和和ab');
+---------------------------+
| reverse('美团和和ab')     |
+---------------------------+
| ba和和团美                |
+---------------------------+
1 row in set (0.01 sec)

mysql> select reverse('美团和和ab');
+---------------------------+
| reverse('美团和和ab')     |
+---------------------------+
| ba和和团美                |
+---------------------------+
1 row in set (0.02 sec)

```

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)
- [ ] Optimization. Including functional usability improvements and performance improvements.
- [ ] Dependency. Such as changes related to third-party components.
- [ ] Other.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have created an issue on (Fix #ISSUE) and described the bug/feature there in detail
- [x] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/functions/function_string.cpp [hunk 1]
```diff
@@ -248,26 +248,34 @@ struct NameReverse {
...
+        // allocate memory and memory align
         res_data.resize(data_length);
-        for (int i = 0; i < size; ++i) {
-            const char* l_raw_str = reinterpret_cast<const char*>(&data[offsets[i - 1]]);
-            int l_str_size = offsets[i] - offsets[i - 1] - 1;
 
-            char* r_raw_str = reinterpret_cast<char*>(&res_data[res_offsets[i - 1]]);
-            memcpy(r_raw_str, l_raw_str, l_str_size);
+        for (int i = 0; i < rows_count; ++i) {
+            const unsigned char* src_str = reinterpret_cast<const unsigned char*>(&data[offsets[i - 1]]);
```
Comment (HappenLee) [inline be/src/vec/functions/function_string.cpp:265]: 这里可以直接auto了，没必要再写一遍类型

File: be/src/vec/functions/function_string.cpp [hunk 2]
```diff
@@ -248,26 +248,34 @@ struct NameReverse {
...
+            // -1 means '\0'
+            size_t src_str_size = offsets[i] - offsets[i - 1] - 1;
+
+            unsigned char* dst_str = reinterpret_cast<unsigned char*>(&res_data[res_offsets[i - 1]]);
```
Comment (HappenLee) [inline be/src/vec/functions/function_string.cpp:269]: 同上

**Comment by zbtzbtzbt** [comment]
刚刚强推吧别人的代码也带上了=。=，有空辛苦再code review一下

**Comment by zbtzbtzbt** [comment]
这个老pr分支我搞乱了，新建了一个，这个先关了

**Pull request closed without merging**